site/source/cookbook/building/edit-dojo-source.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: docs
+category: building
+title: Edit the Dojo 2 source code
+overview: Edit the Dojo 2 source code for experimentation and debugging
+---
+
+## Objective
+
+You may wish to modify source code in Dojo 2 to verify whether something is a bug in your application code, or within Dojo 2. Alternatively, you may wish to make a contribution to the project and assert your changes are working from the perspective of a Dojo 2 application.
+
+This recipe demonstrates how to edit the Dojo 2 source code, and see the change within your app.
+
+## Procedure
+
+1. Download part of the Dojo 2 source code:
+
+```sh
+git clone --depth 1 https://github.com/dojo/widget-core
+```
+
+Tip: `--depth 1` will only grab the latest revision, resulting in a speedier download time.
+
+2. Make the change you require within widget-core, in this example, the `render` invocation within the Dojo 2 framework is instrumented with console timing functions.
+
+Change this code:
+
+```ts
+// src/WidgetBase.ts
+let dNode = render();
+```
+
+Into this:
+
+```ts
+// src/WidgetBase.ts
+console.time('Render');
+let dNode = render();
+console.timeEnd('Render');
+```
+
+3. Build widget-core:
+
+```sh
+cd widget-core
+npm i
+./node_modules/.bin/grunt  release --pre-release-tag=beta3 --dry-run --skip-checks
```

Comment:
I don't think this should be a cook book with the workflow as is right now (this is an internal release command, it's not user friendly and is potentially dangerous). lets raise an issue for fixing the workflow for external users.

Reply:
Oh we don't have to show this command exactly, but is there an alternative? Because how else could an end-user, who wants to contribute/edit the source code, do so? Because wouldn't it be better to show at least one way of achieving this? Also just fyi: I'm not showing a 'contribution' workflow, I'm showing a sort of 'debugging' workflow which I think is a fairly common use case?

Reply:
we need to fix the workflow for external users, we really shouldn't be putting our internal release tool out there as an official way to debug/contribute/anything.

Reply:
ok 👍 have removed the recipe and will make an issue as a reminder to add it in again one day